### ISSUE TITLE
[Question] File extension?

### ISSUE THREAD

**Issue by rmenessec** [isobit/vim-caddyfile#2]
Is there a "common" or "accepted" or "best practice" file extension for Caddyfiles?

I'm not being dogmatic; I just don't see a way to trigger the syntax support except by invoking it manually every time. Am I misreading the code? Do I just need to use ".caddy" or something obvious like that?

@mholt , can you comment?

Thanks in advance.

**Comment by mholt** [comment]
The Caddyfile is usually called just that, `Caddyfile`, with no extension.

**Comment by rmenessec** [comment]
If there's no accepted or common file extension, is there a way to force ViM to remember syntax settings per file? I'm relatively new to ViM plugins, and I'm not clear on whether ViM has any default-shipped facilities of its own to remember per-file settings like this.

Alternately, are there any non-file extension triggers I'm missing that can or should set the Caddyfile syntax support? Something else ViM could use to recognize a Caddyfile without manual intervention?

**Comment by isobit** [comment]
Per the [`ftdetect`](https://github.com/isobit/vim-caddyfile/blob/master/ftdetect/caddyfile.vim), this extension only recognizes files literally called `Caddyfile` (no extension) out of the box.

If you need per-file settings, I suggest using a [modeline](http://vim.wikia.com/wiki/Modeline_magic). Alternatively, you could put an `autocmd` in your `~/.vimrc` to set up custom file type assignment (like the one in the `ftdetect`, but with your own filename).

TL;DR Put `set modeline` in your `vimrc` and put this at the top of your Caddyfile:
```
# vim: set ft=caddyfile:
```

**Comment by isobit** [comment]
Of course, the idiomatic solution would be to just rename your file to "Caddyfile".

**Comment by rmenessec** [comment]
@isobit I just found modeline functionality a few minutes ago. 😉

Your advice on the "Caddyfile" naming is well-taken. Thank you. I didn't see it because I was still looking for a file extension, and the... syntax syntax? ...wasn't making sense until you explained.

Thanks!

And thanks for working on this. It seems to work well for the couple of experimental Caddyfiles I have so far, and looks good with Monokai.

**Comment by rmenessec** [comment]
@isobit wrote:
> Alternatively, you could put an autocmd in your ~/.vimrc to set up custom file type assignment (like the one in the ftdetect, but with your own filename).

Apparently, I'm too much of a vim n00b to do this. I'm being told:
```
E122: Function <SNR>58_sw already exists, add ! to replace it
```
Is there a clean, fast, simple way to override the existing definition? For if not, I will simply go with the modeline suggestion, I think.

**Comment by isobit** [comment]
You should be able to just put the following in your `vimrc`:
```
au BufNewFile,BufRead <YOUR_FILENAME_HERE> set ft=caddyfile
```

Make sure the rest of your `vimrc` doesn't have syntax errors, it sounds like you're defining a function twice or something.

**Comment by rmenessec** [comment]
<sigh> I forgot to remove the modeline. I guess that's what happens when I try to herd vim at 0400 local. Sorry for the confusion. My trigger was identical to your suggestion, except s/ft=/filetype=/

**Comment by isobit** [comment]
What is vim good for if not tinkering until 4am? Hopefully everything's working for you now!

**Comment by rmenessec** [comment]
@isobit It's good for tinkering on _other_ stuff until 4AM. 😀

Yeah, I'm good. The only problem was the modeline that I set as a first resort. Once I clipped that off the top of my one Caddyfile, everything worked correctly.

Thanks for all your help, and for the great syntax plugin!
